vendor/wp-now/src/symlink-manager.ts

```diff
@@ -0,0 +1,220 @@
+			return;
+		}
+
+		const stat = await fs.lstat(fullPath);
+		if( ! stat.isSymbolicLink() ) {
+			return;
+		}
+
+		await this.addSymlink(filename);
+	}
+
+	/**
+	 * Add a symlink to the PHP runtime by mounting the target path.
+	 * @param filename
+	 */
+	async addSymlink(filename: string) {
+
+		const fullPath = path.join(this.projectPath, filename);
+		const target = await fs.realpath(fullPath);
+		const vfsPath = path.join(this.php.requestHandler.documentRoot, filename);
+
+		// Double check to ensure the symlink exists
+		if( ! this.php.fileExists(vfsPath) ) {
+			return;
+		}
+
+		// Get the realpath of the symlink within the PHP runtime.
+		const vfsTarget = this.php.readlink(vfsPath);
+		this.symlinks.set(filename, vfsTarget);
+
+		// Get the mount info for the target
+		let mountInfo = this.mountedTargets.get(vfsTarget);
+
+		// Handle a new mount
+		if (mountInfo === undefined) {
+			mountInfo = {
+				preExisting: this.php.fileExists( vfsTarget ),
+				referenceCount: 0,
+				unmountFunction: null,
+			};
+
+			this.mountedTargets.set(vfsTarget, mountInfo);
+		}
+
+		// Increment the reference count to keep track of how many symlinks are pointing to the target
+		mountInfo.referenceCount++;
+
+		// If the target is already mounted or is pre-existing, we are done
+		if( mountInfo.preExisting || mountInfo.unmountFunction !== null ) {
+			return;
+		}
+
+		// We need to make the target or mounting will fail
+		this.php.mkdir(vfsTarget);
+		mountInfo.unmountFunction = await this.php.mount( vfsTarget, createNodeFsMountHandler( target ) );
+	}
+
+	async deleteMountedTarget(filename: string) {
+		const vfsTarget = this.symlinks.get(filename);
+
+		if (vfsTarget === undefined) {
+			return;
+		}
+
+		// Get the mount info for the target
+		let mountInfo = this.mountedTargets.get(vfsTarget);
+		if (mountInfo === undefined) {
+			return;
+		}
+
+		mountInfo.referenceCount--;
+
+		// If no more symlinks are pointing to the target, we can remove the target from the map
+		if (mountInfo.referenceCount == 0 ) {
+			this.symlinks.delete(filename);
+		}
+
+		// If the reference count is 0 and the unmount function is set, we can unmount the target
+		if (mountInfo.referenceCount === 0 && mountInfo.unmountFunction !== null && !mountInfo.preExisting) {
+			await mountInfo.unmountFunction();
+
+			// After unmounting, clear the mount target
+			if( this.php.isFile(vfsTarget) ) {
+				this.php.unlink(vfsTarget);
+			}
+
+			if( this.php.isDir(vfsTarget) ) {
+				this.php.rmdir(vfsTarget);
+			}
+
+			this.mountedTargets.delete(vfsTarget);
+		}
+	}
+
+	/**
+	 * Stop watching the project path for changes.
+	 */
+	async stopWatching() {
+		if ( !this.watcher ) {
+			return;
+		}
+		this.watcherAbortController.abort('Watcher stopped');
+	}
+
+	/**
+	 * Check if a file exists.
+	 * @param filename
+	 */
+	private async fileExists( filename: string ) {
+		try {
+			await fs.access( filename );
+			return true;
+		} catch( err ) {
+			return false;
+		}
```

Comment:
`fs.access` can fail due to other causes. I wonder if we should only return `false` in case we encounter the "file not found" error.

```suggestion
		try {
			await fs.access(filename, fs.constants.F_OK);
			return true;
		} catch (error) {
			if (error.code === 'ENOENT') {
				return false;
			}
			// Rethrow unexpected errors
			throw error;
		}
```